warehouse/query-core/src/test/java/datawave/query/jexl/visitors/ValidComparisonVisitorTest.java

```diff
@@ -0,0 +1,136 @@
+        String queryString = "BAR == FOO && FOO =~ BAR && BAR != FOO && FOO !~ BAR && BAR < FOO  && FOO > BAR && BAR <= FOO && FOO >= BAR";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(queryString);
+        ValidComparisonVisitor.check(script);
+    }
+    
+    @Test(expected = DatawaveFatalQueryException.class)
+    public void testInvalidEQ() throws ParseException {
+        String queryString = "'BAR' == 1";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(queryString);
+        ValidComparisonVisitor.check(script);
+    }
+    
+    @Test(expected = DatawaveFatalQueryException.class)
+    public void testInvalidNE() throws ParseException {
+        String queryString = "'BAR' != 1";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(queryString);
+        ValidComparisonVisitor.check(script);
+    }
+    
+    @Test(expected = DatawaveFatalQueryException.class)
+    public void testInvalidRE() throws ParseException {
+        String queryString = "'BAR' =~ 'foo'";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(queryString);
+        ValidComparisonVisitor.check(script);
+    }
+    
+    @Test(expected = DatawaveFatalQueryException.class)
+    public void testInvalidNR() throws ParseException {
+        String queryString = "'BAR' !~ 'foo'";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(queryString);
+        ValidComparisonVisitor.check(script);
+    }
+    
+    @Test(expected = DatawaveFatalQueryException.class)
+    public void testInvalidLT() throws ParseException {
+        String queryString = "'BAR' < 1";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(queryString);
+        ValidComparisonVisitor.check(script);
+    }
+    
+    @Test(expected = DatawaveFatalQueryException.class)
+    public void testInvalidLE() throws ParseException {
+        String queryString = "'BAR' <= 1";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(queryString);
+        ValidComparisonVisitor.check(script);
+    }
+    
+    @Test(expected = DatawaveFatalQueryException.class)
+    public void testInvalidGT() throws ParseException {
+        String queryString = "'BAR' > 1";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(queryString);
+        ValidComparisonVisitor.check(script);
+    }
+    
+    @Test(expected = DatawaveFatalQueryException.class)
+    public void testInvalidGE() throws ParseException {
+        String queryString = "'BAR' >= 1";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(queryString);
+        ValidComparisonVisitor.check(script);
+    }
+    
+    @Test(expected = DatawaveFatalQueryException.class)
+    public void testInvalidAssignment() throws ParseException {
+        String queryString = "'ASTEvaluationOnly' = true";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(queryString);
+        ValidComparisonVisitor.check(script);
+    }
+    
```

Comment:
You removed the concept of passing found identifiers back up to parent nodes as I was doing with the list implementation.  Add this test and you'll see how it fails.

```suggestion
    
    @Test(expected = DatawaveFatalQueryException.class)
    public void testMethodNodeNested() throws ParseException {
        String queryString = "(AG.greaterThan(39).size() >= 1) == 'false'";
        ASTJexlScript script = JexlASTHelper.parseJexlQuery(queryString);
        ValidComparisonVisitor.check(script);
    }
    
```

Reply:
I contemplated that, and I don't believe that your added test query should actually fail here.  I am only trying to catch the case where it is a simple literal op literal expression where it is highly likely that the user made an error specifying the query.  If it is something more complicated like this one, then I am more inclined to believe the user knows what they are doing.